src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -53,7 +53,9 @@ case class CsvRelation protected[spark] (
   extends BaseRelation with TableScan with PrunedScan with InsertableRelation {
 
   // Share date format object as it is expensive to parse date pattern.
-  private val dateFormatter = if (dateFormat != null) new SimpleDateFormat(dateFormat) else null
+  private val dateFormatter = if (dateFormat != null) {
+    dateFormat.split(",").map(df => new SimpleDateFormat(df))
+  } else null
```

Comment:
I hope the indentation is fixed as below:

```scala
if (dateFormat != null) {
  dateFormat.split(",").map(df => new SimpleDateFormat(df))
} else {
  null
}
```

and maybe it might better avoid to use name `df` because it is used for `DataFrame` as far as I know.